src/common/mfu_flist_copy.c

```diff
@@ -1682,6 +1738,14 @@ static int mfu_copy_file_normal(
 
         /* add bytes to our total (use bytes read,
          * which may be less than number written) */
+#ifdef DAOS_SUPPORT
+        if (mfu_src_file->type == DAOS) {
+            mfu_src_file->offset += num_of_bytes_read;
+        }
+        if (mfu_dst_file->type == DAOS) {
+            mfu_dst_file->offset += num_of_bytes_read;
+        }
+#endif
```

Comment:
Let's move these offset updates to the daos_read and daos_write functions.  Users calling mfu_file_read and mfu_file_write will expect the file pointers to be updated as part of the call.